payjoin-cli/src/app/v2.rs

```diff
@@ -310,53 +309,27 @@ async fn unwrap_ohttp_keys_or_else_fetch(config: &AppConfig) -> Result<payjoin::
     if let Some(keys) = config.ohttp_keys.clone() {
         Ok(keys)
     } else {
-        fetch_ohttp_keys(&config.ohttp_relay, &config.pj_endpoint).await
+        let ohttp_relay = config.ohttp_relay.clone();
+        let payjoin_directory = config.pj_endpoint.clone();
+        #[cfg(feature = "danger-local-https")]
+        let cert_der = rcgen::generate_simple_self_signed(vec![
+            "0.0.0.0".to_string(),
+            "localhost".to_string(),
+        ])?
+        .serialize_der()?;
+        Ok(tokio::task::spawn_blocking(move || {
+            payjoin_defaults::fetch_ohttp_keys(
+                ohttp_relay,
+                payjoin_directory,
+                #[cfg(feature = "danger-local-https")]
+                cert_der,
+            )
+        })
+        .await
+        .map_err(|e| anyhow!("Failed to fetch ohttp keys: {}", e))??)
```

Comment:
It's a bit over complicated a fallible expression in Ok() without first handling the error. Better to handle one error at a time and then return Ok if the result is actually Ok. We can fix this in a follow up.